### PR TITLE
Improved / Fixed learning rate parameters (WIP)

### PR DESCRIPTION
**Pull request by ExponentialML**
This is a work in progress (I haven't tested the merging of `dicts` yet).

1. This sets the CLIP parameters in line with https://github.com/openai/CLIP/issues/83 for text encoder training
2. Separates parameter groups. Currently :
```python
optimizer_class(params_to_optimize, lr=args.learning_rate if not args.use_lora else args.lora_learning_rate)
```
Can lead to cases where the learning rates aren't being utilized properly when using LoRA. Placing the parameters in separate groups can make things a bit more deterministic. 

Feel free to try and let us know of any improvements.

### CODE REVIEW

**Comment by ExponentialML** [comment]
Closing for now. This is more of a custom use case rather than be in the full repository.
